### PR TITLE
Zehua Week11 hw

### PR DESCRIPTION
**Pull request by zehua**
Please ignore files in `week10` folder as they are part of PR #156 . Once that's merged, I will rebase to clean up the changeset.

* [ ] rebase after #156 is merged

### CODE REVIEW

File: cis194/week11/zehua/src/AParser.hs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+-- Your code goes below here
+------------------------------------------------------------
+
+-- ex1
+first :: (a -> b) -> (a, c) -> (b, c)
+first f (a, c) = (f a, c)
+
+instance Functor Parser where
+  -- fmap :: (a -> b) -> Parser a -> Parser b
+  -- fmap :: (a -> b) -> (String -> Maybe (a, String)) -> (String -> Maybe (b, String))
```
Comment (zehua) [inline cis194/week11/zehua/src/AParser.hs]: this is technically not correct. just expanding it out to help understand the type

File: cis194/week11/zehua/src/AParser.hs [hunk 2]
```diff
@@ -0,0 +1,157 @@
+module AParser where
```
Comment (zehua) [inline cis194/week11/zehua/src/AParser.hs]: This file is copied from week10.

Comment (zehua) [reply cis194/week11/zehua/src/AParser.hs]: changed to symlink

**Comment by zehua** [comment]
@shineli1984 PR rebased

**Review by shineli1984 — APPROVED** [review]

**Pull request merged**